### PR TITLE
Add support for NSData-based PDF renderers (and code cleanup)

### PR DESCRIPTION
**Pull request by myell0w**
# Changes proposed in this PR
- Convert testing framework to XCTestCase
- Bump deployment target to 7.0
- Turn on several warnings
- Add support for NSData-based PDF renderers
# Tests performed
- Cmd + U
- creation of PDF renderer with NSURL
- creation of PDF renderer with NSData
# Further info for the reviewer

I've changed some code styling as well as I plan to maintain the Pod based on our needs and it's easier for me to stick to a single code style, let me know what you think of that or whether you dislike it.
# Open Issues

No known issues.

### CODE REVIEW

File: Source/FTPDFAssetRenderer.m [hunk 1]
```diff
@@ -1,145 +1,220 @@
...
+    return nil;
 }
 
 @end
 
 
 @interface FTPDFAssetRenderer () {
-  CGPDFDocumentRef _document;
+    CGPDFDocumentRef _document;
+    NSString *_cachePathIdentifier;
```
Comment (alloy) [inline Source/FTPDFAssetRenderer.m:34]: These should really just become properties on a private class extension nowadays imo, that should probably also take care of memory management.

**Comment by alloy** [comment]
Looks great 👍 

My only real style gripe is with the `self = [super init]` parts, but seeing as it’s more likely that you will work on it in the near future, I don’t consider it a blocker.

**Comment by alloy** [comment]
As a refactor idea, I wonder if it might not be easier to make the data loading variant the designated initialiser and make the URL variant load data from that URL.

But I don’t know from the top of my head if CGPDFDocument might be more efficient in data loading, although otoh these PDFs are typically going to be small asset ones.

**Comment by alloy** [comment]
I’m going to give you push access, once I’ve heard back from @Manfred about whether to keep it on the Fingertips account or move it elsewhere.

Shall I tag this as v1, allowing you the freedom to change API further for a v2?

**Comment by Manfred** [comment]
Thanks for the changes Matthias!

@alloy I think it's fine to leave it on our organization as long as there are only a few developers working on it.

**Comment by alloy** [comment]
Done.

**Pull request merged**